website/views.py

```diff
@@ -214,10 +215,10 @@ def resolve_guid(guid, suffix=None):
         if not referent.deep_url:
             raise HTTPError(http.NOT_FOUND)
         if isinstance(referent, PreprintService):
-            return send_from_directory(
-                os.path.abspath(os.path.join(os.getcwd(), EXTERNAL_EMBER_APPS['preprints']['path'])),
-                'index.html'
-            )
+            if referent.provider._id == 'osf' or not referent.provider.domain:
```

Comment:
I'm not 100% clear how time-sensitive this PR is, but GUIDs are complex, and this code adds some new special cases. Normally, we would encourage unit tests for such functionality as time permits. 

Let me know if I can be of assistance in finding the right place to test that and move forward. I might consider some tests on the model (where url behavior is conditional), but that's not as high a priority as testing the GUID functionality.